experimental/immediate-accept-adapter/csharp_dotnetcore/ImmediateAcceptAdapter.cs

```diff
@@ -0,0 +1,127 @@
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Integration.AspNet.Core;
+using Microsoft.Bot.Builder.TraceExtensions;
+using Microsoft.Bot.Connector.Authentication;
+using Microsoft.Bot.Schema;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+
+namespace ImmediateAcceptBot
+{
+    public class ImmediateAcceptAdapter : BotFrameworkHttpAdapter
+    {
+        private readonly IBackgroundTaskQueue _backgroundTaskQueue;
+        private readonly IActivityTaskQueue _activityTaskQueue;
+
+        public ImmediateAcceptAdapter(IConfiguration configuration, ILogger<BotFrameworkHttpAdapter> logger, IActivityTaskQueue activityTaskQueue, IBackgroundTaskQueue backgroundTaskQueue)
+            : base(configuration, logger)
+        {
+            _backgroundTaskQueue = backgroundTaskQueue;
+            _activityTaskQueue = activityTaskQueue;
+
+            OnTurnError = async (turnContext, exception) =>
+            {
+                // Log any leaked exception from the application.
+                logger.LogError(exception, $"[OnTurnError] unhandled error : {exception.Message}");
+
+                // Send a message to the user
+                await turnContext.SendActivityAsync("The bot encountered an error or bug.");
+                await turnContext.SendActivityAsync("To continue to run this bot, please fix the bot source code.");
+
+                // Send a trace activity, which will be displayed in the Bot Framework Emulator
+                await turnContext.TraceActivityAsync("OnTurnError Trace", exception.Message, "https://www.botframework.com/schemas/error", "TurnError");
+            };
+        }
+
+        /// <summary>
+        /// This method can be called from inside a POST method on any Controller implementation.  If the activity is Not an Invoke, and
+        /// DeliveryMode is Not ExpectReplies, and this is not a Get request to upgrade to WebSockets, then the activity will be enqueued
+        /// to be processed on a background thread.
+        /// 
+        /// 
+        /// 
+        /// Note, this is an ImmediateAccept and BackgroundProcessing replacement for: 
+        /// Task ProcessAsync(HttpRequest httpRequest, HttpResponse httpResponse, IBot bot, CancellationToken cancellationToken = default);
+        /// </summary>
+        /// <param name="httpRequest">The HTTP request object, typically in a POST handler by a Controller.</param>
+        /// <param name="httpResponse">The HTTP response object.</param>
+        /// <param name="bot">The bot implementation.</param>
+        /// <param name="cancellationToken">A cancellation token that can be used by other objects or threads to receive
+        ///     notice of cancellation.</param>
+        /// <returns>A task that represents the work queued to execute.</returns>
+        public async Task ProcessOnBackgroundThreadAsync(HttpRequest httpRequest, HttpResponse httpResponse, IBot bot, CancellationToken cancellationToken = default)
```

Comment:
Do you need a different method here?
I mean you could just override and delegate calls to the base class appropriately.
(In other words the relationship with the Controller - which is what the existing interface is modeling - technically hasn't changed - so I don't need a new type exposed to the Controller. OK I realize the semantics have changed slightly. But the semantics of the protocol are a little unconventional - and not directly modeled - anyhow.)

Reply:
@johnataylor  Should we make ProcessAsync virtual or protected?

Currently:
```cs
public Task ProcessAsync(HttpRequest httpRequest, HttpResponse httpResponse, IBot bot, CancellationToken cancellationToken = default);
```